src/libmongoc/tests/test-mongoc-topology.c

```diff
@@ -2305,6 +2305,146 @@ test_hello_versioned_api_pooled ()
    _test_hello_versioned_api (true);
 }
 
+static void
+_test_hello_ok (bool pooled)
```

Comment:
This test is great! Would it be reasonable to extend this test and parameterize on whether an API version is set?

That seems impactful to test. Since the legacy hello command is not part of server API, if monitoring accidentally sends the legacy hello command in an error scenario, it may not be able to recover.

Reply:
`_test_hello_versioned_api` in L2227 tests exactly that: it declares an Api Version on the client and checks if the first handshake is using the `hello` command and not legacy hello.